test_unstructured/metrics/test_text_extraction.py

```diff
@@ -59,11 +98,47 @@ def test_calculate_edit_distance_with_filename(filename, expected_score, expecte
     elements = partition(filename=f"example-docs/{filename}")
     output_cct = "\n".join([str(el) for el in elements])
 
-    score = calculate_edit_distance(output_cct, source_cct, return_as="score")
-    distance = calculate_edit_distance(output_cct, source_cct, return_as="distance")
+    score = text_extraction.calculate_edit_distance(output_cct, source_cct, return_as="score")
+    distance = text_extraction.calculate_edit_distance(output_cct, source_cct, return_as="distance")
 
     assert score >= 0
     assert score <= 1.0
     assert distance >= 0
     assert round(score, 2) == expected_score
     assert distance == expected_distance
+
+
+@pytest.mark.parametrize(
+    ("text", "expected"),
+    [
+        (
+            "The dog loved the cat, but the cat loved the cow",
+            {"the": 4, "cat": 2, "loved": 2, "dog": 1, "but": 1, "cow": 1},
+        ),
+        (
+            "Hello my name is H a r p e r, what's your name?",
+            {"hello": 1, "my": 1, "name": 2, "is": 1, "what's": 1, "your": 1},
+        ),
+        (
+            "I have a dog and a cat, I love my dog.",
+            {"i": 2, "have": 1, "a": 2, "dog": 2, "and": 1, "cat": 1, "love": 1, "my": 1},
+        ),
+        (
+            "My dog's hair is red, but the dogs' houses are blue.",
+            {
+                "my": 1,
+                "dog's": 1,
+                "hair": 1,
+                "is": 2,
+                "red": 1,
+                "but": 1,
+                "the": 1,
+                "dog's": 1,
```

Comment:
This should be `dogs'` with the apostrophe at the end (freq 1), since `dog's` is already its own category (freq 1)